### DISCUSSION TITLE
Fails of "test-render debug/raster" and "test-render debug/tile"

### DISCUSSION THREAD

**Discussion by davychxn** [maplibre/maplibre-gl-js#1759] [General]
Hi,

The following tests basically fail on fonts drawing:
1. test-render debug/raster
2. test-render debug/tile

Take "test-render debug/raster" as an example:
Actual:
![actual](https://user-images.githubusercontent.com/3283265/197059250-82450ad5-2165-42c0-bbce-ee4f02412baf.png)
Expected:
![expected](https://user-images.githubusercontent.com/3283265/197059294-8239b0a1-1741-415d-8462-f9a8341bd72b.png)
Diff:
![diff](https://user-images.githubusercontent.com/3283265/197059321-be05fccf-5811-4b6b-937c-5e1c51c56ebe.png)

The fonts are drawn by code here:
"https://github.com/maplibre/maplibre-gl-js/blob/main/src/render/draw_debug.ts"
![image](https://user-images.githubusercontent.com/3283265/197059815-2bc3022f-da69-4422-bd51-b1c24d5571aa.png)

Form history, above code wasn't changed big.

On May of 2022 exact the 2 tests were updated once:
![image](https://user-images.githubusercontent.com/3283265/197060282-2e72e6d1-1866-4472-b7bf-e71b73b18c85.png)

At that time the tests should still passed.

At that time "gl" (headless-gl) package was still at version 5.0.0

I did following tests:
1, I reverted this history change: https://github.com/maplibre/maplibre-gl-js/commit/cfbc39f8316c9624256df48ffff35c3f0f629ea3

Manually replace "gl" to version 5.0.0
Built and run the tests, still fail.

2, Downloaded "dist" folder of "https://github.com/maplibre/maplibre-gl-js/releases/tag/v2.2.0-pre.1".
Which was released on May 12, 2022. Run the tests, still fail.

Since the fonts are rendered using Canvas calls, I think the fonts rendering result's difference is caused by "head-less".

Can I just update the "expected.png" to latest? Or any suggestion for further investigation?
 
Thanks.

**Comment by HarelM** [comment]
There are two options here:
1. Update the expected.png to pass for you and for the CI - preferred
2. If 1 isn't possible you can add another expected.png inn that folder, the tests will try to find the expected with the lowest diff.
